src/libraries/System.Diagnostics.Process/tests/ProcessWaitingTests.cs

```diff
@@ -82,6 +154,43 @@ public void SingleProcess_TryWaitMultipleTimesBeforeCompleting()
             Assert.True(p.WaitForExit(0));
         }
 
+        [Fact]
+        public async Task SingleProcess_TryWaitAsyncMultipleTimesBeforeCompleting()
+        {
+            Process p = CreateProcessLong();
+            p.Start();
+
+            // Verify we can try to wait for the process to exit multiple times
+            using (var cts = new CancellationTokenSource(millisecondsDelay: 0))
+            {
+                var token = cts.Token;
+                for (int i = 0; i < 2; i++)
+                {
+                    OperationCanceledException ex = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => p.WaitForExitAsync(token));
```

Comment:
You're passing in a token that's already been canceled by the time you call this.  As such, WaitForExitAsync should actually complete synchronously, and it'd be better to lock that behavior into the test, e.g.
```C#
Task t = process.WaitForExitAsync(token);
Assert.Equal(TaskStatus.Canceled, t.Status);
OperationCanceledException ex = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => t);
```

Reply:
Added!